google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -134,6 +134,18 @@ public static CloudStorageFileSystem forBucket(String bucket) {
     return forBucket(bucket, userSpecifiedDefault);
   }
 
+  /**
+   * Creates new file system instance for {@code bucket}, with existing provider and configuration.
+   *
+   * @param bucketName name of the bucket to initialize {@code CloudStorageFileSystem} object
+   * @return {@code CloudStorageFileSystem} object with existing provider and config
+   * @see #forBucket(String)
+   */
+  private static CloudStorageFileSystem getExistingCloudStorageConfiguration(String bucketName) {
+    return new CloudStorageFileSystem(
+        cloudStorageFileSystem.provider(), bucketName, cloudStorageFileSystem.config());
+  }
+
```

Comment:
Method name `getExistingCloudStorageConfiguration` implies returning *existing configuration*, you return FileSystem and create it each time.

Reply:
Done